ChopChop/Parser/RecipeParser.swift

```diff
@@ -0,0 +1,224 @@
+            + optionalWhiteSpace + groupWithName(pattern: units, name: "unit", isOptional: true)
+            + whitespace + groupWithName(pattern: ".*", name: "ingredient")
+
+        let regex = NSRegularExpression(pattern)
+        guard let result =
+            regex.firstMatch(in: text, options: [], range: NSRange(location: 0, length: text.utf16.count)) else {
+
+            return nil
+        }
+
+        guard let numeralRange = Range(result.range(withName: "numeral"), in: text),
+              let ingredientRange = Range(result.range(withName: "ingredient"), in: text) else {
+
+            print("Incorrect match")
+            return nil
+        }
+
+        var value: Double = 0
+        let numberText = text[numeralRange]
+        if isFraction(text: String(numberText)) {
+            value = parseFraction(fraction: String(numberText))
+        } else if let number = Double(numberText) {
+            value = number
+        } else {
+            print("Incorrect match")
+            return nil
+        }
+
+        var quantity: Quantity?
+        if let unitRange = Range(result.range(withName: "unit"), in: text) {
+            let unit = text[unitRange]
+            quantity = convertToQuantity(value: &value, unit: String(unit))
+        } else {
+            quantity = .count(value)
+        }
+
+        let ingredient = text[ingredientRange].trimmingCharacters(in: .whitespaces)
+
+        guard let ingredientQuantity = quantity, !ingredient.isEmpty else {
+            print("Invalid quantity or ingredient")
+            return nil
+        }
+
+        return (ingredient, ingredientQuantity)
+    }
+
+    private static func parseFraction(fraction: String) -> Double {
+        let fractionArray = fraction.split(separator: "/")
+        if fractionArray.count != 2 {
+            print("Incorrect fraction format")
+            return 0
+        }
+        let numeratorStr = fractionArray[0].trimmingCharacters(in: .whitespaces)
+        let denominatorStr = fractionArray[1].trimmingCharacters(in: .whitespaces)
+
+        guard let numerator = Double(numeratorStr), let denominator = Double(denominatorStr) else {
+            print("Incorrect fraction format")
+            return 0
+        }
+
+        return numerator / denominator
+    }
+
+    private static func isFraction(text: String) -> Bool {
+        NSRegularExpression(fraction).matches(text)
+    }
+
+    private static func convertToQuantity( value: inout Double, unit: String) -> Quantity {
+        if let volume = UnitsMapping.volumeWordMap[unit.lowercased()], let factor = UnitsMapping.volumeToL[volume] {
+            value *= factor
+            return .volume(value)
+        } else if let mass = UnitsMapping.massWordMap[unit.lowercased()], let factor = UnitsMapping.massToKg[mass] {
+            value *= factor
+            return .mass(value)
+        } else {
+            // temp
+            return .count(1)
```

Comment:
Is there a reason for doing this instead of returning `.count(value)`?
Especially since earlier there was something like
```
if let unitRange = Range(result.range(withName: "unit"), in: text) {
    let unit = text[unitRange]
    quantity = convertToQuantity(value: &value, unit: String(unit))
} else {
    quantity = .count(value)
}
```

Perhaps make the `unit` parameter optional, then when its `nil` it just means the quantity is a `.count`

Reply:
Yea we can do .count(value), though technically the code shldnt reach the last else block cos if this method is called, it should imply that a unit inside the UnitMapping is found.